Kudu.Services/Diagnostics/ProfileManager.cs

```diff
@@ -83,30 +128,58 @@ internal static class ProfileManager
                     profilingSessionId = _profilingList[processId].SessionId;
                 }
 
-                var profileProcessResponse = await StopProfileInternalAsync(processId, profilingSessionId, false, tracer);
+                var profileProcessResponse = await StopProfileInternalAsync(processId, profilingSessionId, false, tracer, iisProfiling);
 
                 return profileProcessResponse;
             }
         }
 
-        internal static string GetProfilePath(int processId)
+        internal static string GetProfilePath(int processId, bool iisProfiling)
         {
-            string profileFileName = string.Format("profile_{0}_{1}_{2}.diagsession", InstanceIdUtility.GetShortInstanceId(), System.Diagnostics.Process.GetProcessById(processId).ProcessName, processId);
+            string profileFileName = "";
+            
+            if (iisProfiling)
+            {
+                profileFileName = string.Format("profile_{0}_{1}_{2}_{3}.diagsession", InstanceIdUtility.GetShortInstanceId(), "IISProfiling", System.Diagnostics.Process.GetProcessById(processId).ProcessName, processId);
+            }
+            else
+            {
+                profileFileName = string.Format("profile_{0}_{1}_{2}.diagsession", InstanceIdUtility.GetShortInstanceId(), System.Diagnostics.Process.GetProcessById(processId).ProcessName, processId);
+            }
             return System.Environment.ExpandEnvironmentVariables("%LOCAL_EXPANDED%\\Temp\\" + profileFileName);
         }
 
         internal static bool IsProfileRunning(int processId)
         {
             return _profilingList.ContainsKey(processId);
         }
+        internal static bool IsIISProfileRunning(int processId)
+        {
+            bool iisProfiling = false;
+            if (_profilingList != null)
+            {
+                ProfileInfo profilingId;
+                _profilingList.TryGetValue(processId, out profilingId);
+
+                if (profilingId != null)
+                    iisProfiling = profilingId.IsIISProfiling;
+
```

Comment:
nit: the one liner needs still be in curly brackets. 

if (profilingId != null)
{
          iisProfiling = profilingId.IsIISProfiling;
}